server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -758,11 +762,11 @@ private boolean verifyManifestAndClusterState(ClusterMetadataManifest manifest,
             return true;
         }
 
-        private boolean shouldWriteFullClusterState(ClusterState clusterState) {
+        private boolean shouldWriteFullClusterState(ClusterState clusterState, int codecVersion) {
             if (lastAcceptedState == null
                 || lastAcceptedManifest == null
                 || lastAcceptedState.term() != clusterState.term()
-                || lastAcceptedManifest.getOpensearchVersion() != Version.CURRENT) {
+                || lastAcceptedManifest.getCodecVersion() != codecVersion) {
```

Comment:
lets add assert that lastAcceptedManifest.getCodecVersion() <= codecVersion

Reply:
once we move to higher codec version, it can't move back to lower codec version automatically. Either this should be a static setting passed explicitly to cluster manager. There is a risk of creating these as dynamic setting as it can mess up with cluster state.

Reply:
Added assert